File: solution files/Javascript/part-05/task-02-structured-logging/structuredLogging.js
/**
 * Emit a structured audit log. Never include actual secret values.
 * @param {string} url
 * @param {number} secretsCount
 * @param {boolean} success
 * @param {string} [reason]
 */
function logMissionResult(url, secretsCount, success, reason = "") {
    console.log(
        JSON.stringify({
            timestamp: new Date().toISOString(),
            source: url,
            secrets_recovered: secretsCount,
            success,
            reason,
        })
    );
}

// Good - logs metadata only:
logMissionResult("https://raw.githubusercontent.com/...", 5, true);

module.exports = { logMissionResult };

File: solution files/Python/part-05/task-02-structured-logging/structured_logging.py
import json
import logging
from datetime import datetime, timezone


def log_mission_result(url: str, secrets_count: int, success: bool, reason: str = "") -> None:
    """Emit a structured audit log entry. Never log secret content."""
    entry = {
        "timestamp": datetime.now(timezone.utc).isoformat(),
        "source": url,
        "secrets_recovered": secrets_count,
        "success": success,
        "reason": reason,
    }
    logging.info(json.dumps(entry))


if __name__ == "__main__":
    logging.basicConfig(level=logging.INFO)
    log_mission_result("https://raw.githubusercontent.com/...", secrets_count=5, success=True)
